libs/community/langchain_community/chat_message_histories/sql.py

```diff
@@ -123,18 +184,85 @@ def messages(self) -> List[BaseMessage]:  # type: ignore
                 messages.append(self.converter.from_sql_model(record))
             return messages
 
+    async def aget_messages(self) -> List[BaseMessage]:
+        """Retrieve all messages from db"""
+        assert self.async_mode, "This method must be called with async_mode"
+        await self._acreate_table_if_not_exists()
+        async with self.session_maker() as session:
```

Comment:
Could you change this to create an async session maker, and place relevant asserts inside it?

```python
async with self.asession_maker() as session:
```

Reply:
No. Because the 
```
with self.session_maker() as session:
```
is different with
```
async with self.session_maker() as session:
```

Reply:
And, the self.session_maker is a *variable*. It's may be changed by the user. If the user changes the value, the assertion is no longer present